ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -57,10 +58,10 @@ pub(crate) struct Module {
     ///
     /// An [ExtraArgsIdx] describes an index into this.
     extra_args: Vec<Operand>,
-    /// The constant table.
+    /// The constant pool.
     ///
     /// A [ConstIdx] describes an index into this.
-    consts: TiVec<ConstIdx, Constant>,
```

Comment:
The commit message says:
>  We previously used a `Vec` (well, a `TiVec`) in a manner that was effectively a set (e.g. using `position`)

A set is unordered, so it makes little sense to speak of the position of an element in a set. The constant table is a mapping.

What I think this change conceptually does is makes insertion happen by hashing the value (not the key, as in a traditional mapping), thus removing the need to do searches for pre-existing values upon insertions, right?

If so, great! I just disagree with the commit message.